Sources/ClientRuntime/Plugins/Plugin.swift

```diff
@@ -5,6 +5,8 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-public protocol Plugin {
-    func configureClient(clientConfiguration: ClientConfiguration) async throws
+public protocol Plugin<Config> {
+    associatedtype Config: ClientConfiguration
+
+    func configureClient(clientConfiguration: inout Config) async throws
```

Comment:
`Plugin` now takes an `inout clientConfiguration` so that it may make permanent changes to the passed config.

The type of the config is now generic to the plugin so that additional casting is not needed on the config in order to access its properties.